lib/rack/idempotency/request.rb

```diff
@@ -1,14 +1,20 @@
 require "rack/request"
+require "backports/2.4.0/string/match" if RUBY_VERSION < "2.4.0"
 
 module Rack
-  class Idempotency    
+  class Idempotency
     class Request < Rack::Request
       def idempotency_key
-        get_header("HTTP_IDEMPOTENCY_KEY").tap do |key|
-          unless key.nil? || key.match?(/^[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$/i)
-            raise InsecureKeyError.new(env), 'Idempotency-Key must be a valid UUID'
-          end
+        key = if Rack.version > '2.0.0'
```

Comment:
1) Shouldn't this be `Rack.version >= '2.0.0'`

2) We should try to interfere with the code as little as possible when back porting. Does Rack < 2.0.0 have a `get_header` method, but it behaves differently? Or does it not have a `get_header` method at all?

If it does not have a `get_header` method at all, why not just provide one? This way, you don't have to modify the original code at all:

```ruby
def get_header(header_name)
  @env[header_name]
end
```

Reply:
@yemartin as I thought, did fix in other pull request, cherry-picked here now.